src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -1017,34 +1028,180 @@ void EmitAllBranches()
 
                     // Successfully completed the alternate.
                     MarkLabel(doneAlternateLabel);
-                    writer.WriteLine(";");
                     Debug.Assert(textSpanPos == 0);
                 }
             }
 
+            // Emits the code to handle a backreference.
+            void EmitBackreference(RegexNode node)
+            {
+                int capnum = RegexParser.MapCaptureNumber(node.M, rm.Code.Caps);
+
+                TransferTextSpanPosToRunTextPos();
+
+                using (EmitBlock(writer, $"if (base.IsMatched({capnum}))"))
+                {
+                    string matchLength = ReserveName("matchLength");
+                    writer.WriteLine($"int {matchLength} = base.MatchLength({capnum});");
+                    using (EmitBlock(writer, $"if ({textSpanLocal}.Length < {matchLength})"))
+                    {
+                        writer.WriteLine($"goto {doneLabel};");
+                    }
+                    writer.WriteLine();
+
+                    string matchIndex = ReserveName("matchIndex");
+                    writer.WriteLine($"int {matchIndex} = base.MatchIndex({capnum});");
+
+                    string i = ReserveName("i");
+                    using (EmitBlock(writer, $"for (int {i} = 0; {i} < {matchLength}; {i}++)"))
+                    {
+                        using (EmitBlock(writer, $"if ({ToLowerIfNeeded(hasTextInfo, options, $"runtext[{matchIndex} + {i}]", IsCaseInsensitive(node))} != {ToLowerIfNeeded(hasTextInfo, options, $"{textSpanLocal}[{i}]", IsCaseInsensitive(node))})"))
```

Comment:
So this is the case where in the future we will just look at the tables at match time in case IgnoreCase is being used. Does that mean that the case folding tables will also be part of the generated code, or where you thinking to just have those in S.T.RE.dll and just call them from here?

Reply:
> Does that mean that the case folding tables will also be part of the generated code, or where you thinking to just have those in S.T.RE.dll and just call them from here?

I think we'll need to emit the table (or some form of it) into the source _iff_ this code is used, meaning you have an IgnoreCase backreference; we would only need to emit it once per assembly.  The alternative would be to expose a new protected API from RegexRunner for this to call into, which is also an option.